builder/comp-builder.nix

```diff
@@ -95,11 +99,23 @@ let
       (enableFeature enableExecutableProfiling "executable-profiling")
       (enableFeature enableStatic "static")
       (enableFeature enableShared "shared")
+    ] ++ lib.optionals (stdenv.hostPlatform.isMusl && (haskellLib.isExecutableType componentId)) [
+      # These flags will make sure the resulting executable is statically linked.
+      # If it uses other libraries it may be nicessary for to add more
+      # `--ghc-option=-optl=-L` options to the `configurationFlags` of the
+      # component.
+      "--disable-executable-dynamic"
+      "--ghc-option=-optl=-pthread"
+      "--ghc-option=-optl=-static"
+      "--ghc-option=-optl=-L${gmp.override { withStatic = true; }}/lib"
     ] ++ lib.optional enableSeparateDataOutput "--datadir=$data/share/${ghc.name}"
       ++ lib.optional doHaddock' "--docdir=${docdir "$doc"}"
       ++ lib.optional (enableLibraryProfiling || enableExecutableProfiling) "--profiling-detail=${profilingDetail}"
       ++ lib.optional stdenv.hostPlatform.isLinux (enableFeature enableDeadCodeElimination "split-sections")
-      ++ lib.optionals (stdenv.hostPlatform != stdenv.buildPlatform) (
+      ++ lib.optionals (
+                stdenv.hostPlatform != stdenv.buildPlatform
+            # When building for musl on a linux system we do not need to use the `--cross-compile` flag
+            && !(stdenv.buildPlatform.isLinux && stdenv.hostPlatform.isMusl)) (
```

Comment:
This is the inverse logic of the one above  L29-L31, should we extract that into `crossCompiler` with sufficient comments?